### PR TITLE
Firefox: Removing SDK extension tests and generalizing selenium tests…

### PR DESCRIPTION
**Pull request by Hainish**
… to support FF

### CODE REVIEW

File: test/firefox/test_profile_skeleton/prefs.js [hunk 1]
```diff
@@ -1,20 +1,9 @@
-// Submit SSL Observatory reports.
-user_pref("extensions.https_everywhere._observatory.enabled", true);
-// Don't show any popups that might get in the way of testing.
-user_pref("extensions.https_everywhere._observatory.popup_shown", true);
-user_pref("extensions.https_everywhere.toolbar_hint_shown", true);
-// Show all logs.
-user_pref("extensions.https_everywhere.LogLevel", 0);
-user_pref("extensions.https_everywhere.log_to_stdout", true);
 // Make it quicker to make manual config changes.
```
Comment (jsha) [inline test/firefox/test_profile_skeleton/prefs.js]: Why is this file only partially deleted?

File: test/script.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python2.7
+#
+# Run Chromium tests for HTTPS Everywhere
```
Comment (jsha) [inline test/script.py]: s/Chromium/Selenium/

File: test/script.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+if sys.argv[1] == "Firefox":
+    fp = webdriver.FirefoxProfile(sys.argv[2])
+    try:
+        if len(sys.argv) == 4:
+            binary = FirefoxBinary(sys.argv[3])
+            driver = webdriver.Firefox(fp, firefox_binary=binary, log_path=os.devnull)
+        else:
+            driver = webdriver.Firefox(fp, log_path=os.devnull)
+        # Allow the extension time to load
```
Comment (jsha) [inline test/script.py]: This comment should be right above the time.sleep. But it's a little alarming: HTTPS Everywhere should ideally redirect all relevant requests. If requests that happen early in the browser load skip the redirects, that's a bug. It looks like Chrome doesn't have an equivalent sleep. Do you know why this is necessary in Firefox?

File: test/script.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+print ''
+
+driver.get('http://freerangekitten.com')
+
+test_failed = False
+if driver.current_url.startswith('https'):
+    print bcolors.OKGREEN + sys.argv[1] + ": HTTP to HTTPS redirection successful" + bcolors.ENDC
+elif driver.current_url.startswith('http'):
+    print bcolors.FAIL + sys.argv[1] + "Chromium: HTTP to HTTPS redirection failed" + bcolors.ENDC
```
Comment (jsha) [inline test/script.py]: s/Chromium: //

File: test/firefox/test_profile_skeleton/prefs.js (hunk 1 above)
Comment (Hainish) [inline test/firefox/test_profile_skeleton/prefs.js]: It's been a while since this commit was made, I'm honestly not sure why.

Comment (Hainish) [inline test/firefox/test_profile_skeleton/prefs.js]: Ah, these are the prefs that are only relevant for the XPCOM extension.  The other prefs are necessary for debugging, installing the extension without a valid signature, etc.

Comment (Hainish) [inline test/firefox/test_profile_skeleton/prefs.js]: These changes belong in https://github.com/EFForg/https-everywhere/pull/11791, I'll move them over.

**Pull request merged**